### PR TITLE
DAOS-16096 test: Add retry loop for comparing free pool space

### PR DESCRIPTION
**Pull request by phender**
Loop retrying the check for the pool free space after destroying half of the containers.  If the check doesn't pass within 60 seconds, then fail the test.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_dfuse_root_container
Skip-func-hw-test-medium-md-on-ssd: false

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by phender** [comment]
Test passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15289/2/testReport/FTEST_dfuse/

File: src/tests/ftest/dfuse/root_container.py [hunk 1]
```diff
@@ -142,21 +143,28 @@ def verify_create_delete_containers(self, pool, device, cont_count, hosts, mount
...
+            pool_space_after <= expected,
+            f"Failed pool free space check after creating {cont_count} containers")
 
         self.log_step(f"Destroy half of the {cont_count} new sub containers ({cont_count // 2})")
         for _ in range(cont_count // 2):
             containers[-1].destroy(1)
             containers.pop()
 
+        self.log.info("Wait 60 seconds before checking for free space gain")
+        time.sleep(60)
```
Comment (daltonbohning) [inline src/tests/ftest/dfuse/root_container.py:156]: I know this was a suggestion from triage but FYI this is just arbitrary and could take longer.

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
A total of 10 passes of the test passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15289/4/testReport/FTEST_dfuse/.  In all 10 cases the verification of the pool space after destroying 1/2 of the containers completed in one loop, e.g.
```
2024-10-18 00:17:01,237 test             L0471 INFO | ==> Step 12: Verify the pool free space >= 23407660496 after destroying half of the containers [elapsed since last step: 70.78s]
2024-10-18 00:17:01,237 root_container   L0160 DEBUG| Check if the pool free space >= 23407660496 - loop 1
2024-10-18 00:17:01,237 test_utils_pool  L0831 INFO | Querying pool TestPool_1
2024-10-18 00:17:01,238 root_container   L0218 INFO |   Current pool free space:   23623651984
2024-10-18 00:17:01,238 root_container   L0219 INFO |   Expected pool free space:  23407660496
2024-10-18 00:17:01,238 root_container   L0168 INFO | Test passed
2024-10-18 00:17:01,238 test             L0471 INFO | ==> Step 13: Create 5 pools with 10 containers mounted under the root container and insert files into each new container [elapsed since last step: 0.00s]
```

File: src/tests/ftest/dfuse/root_container.py [hunk 2]
```diff
@@ -138,25 +139,33 @@ def verify_create_delete_containers(self, pool, device, cont_count, hosts, mount
...
+            loop += 1
+            self.log.debug(
+                "Check if the pool free space >= %s (loop %s/%s)", expected, loop, max_loops)
+            current = self._get_pool_free_space(pool, device, expected)
+            if current >= expected:
+                break
+            if loop >= max_loops:
+                self.fail(
+                    f"Pool free space less than {expected} after destroying half of the containers")
+            time.sleep(int(60 / max_loops))
```
Comment (daltonbohning) [inline src/tests/ftest/dfuse/root_container.py:168]: FYI `sleep` does work with floats
```
>>> import time
>>> time.sleep(1.12345)
>>>
```

And also FYI if you just want to make sure it's a whole number you can use `//`. But the final type depends on the types passed in, so not always an int
```
>>> 123 / 4
30.75 # float
>>> 123 // 4
30 # int
>>> int(123 / 4)
30 # int
>>> 123.0 // 4
30.0 # float!
```

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
10 repetitions of the test passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15289/6/testReport/FTEST_dfuse/RootContainerTest/

**Comment by phender** [comment]
Note: manually update comment and description.

**Review by mjean308 — APPROVED** [review]

**Pull request merged**